src/bm.h

```diff
@@ -653,7 +658,7 @@ void basm_push_deferred_operand(Basm *basm, Inst_Addr addr, String_View label)
 Word number_literal_as_word(String_View sv)
 {
     assert(sv.count < 1024);
-    char cstr[sv.count + 1];
+    char cstr[1024];
```

Comment:
This is not the intended behaviour.

If we know the String_View holds a string with sv.count characters, we need one more character as terminating '\0'.
Your change allocates 1K, even though `sv` may only hold 20 chars.

Please explain the reasoning behind this subtle change.

Reply:
Btw, that magic number `1024` should be #defined in `bm.h`, something like:
```c
#define BM_STRING_CAPACITY 1024
```
... and used here as
```c
assert(sv.count < BM_STRING_CAPACITY);
char cstr[BM_STRING_CAPACITY];
```

Reply:
Thanks for pointing that out!
It's because of MSVC, it doesn't know the size of the array. And they don't want to add support for VLAs. It's optional for C11.
`sv` may hold any number of characters, because after attempt to parse `ull` we then try to parse a `float`.
For example, this is a valid input as `123f.basm`:
```asm
push 1.0
push 2.0
push 123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789.12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123
plusf
plusf
halt
```